woocommerce/woocommerce-ios
# Jetpack Setup: Fix issue handling completion when signing in with site credentials

Part of #7370


### Description

This PR handles the completion of Jetpack Setup when the flow is started during site credential logins. The solution is simply displaying a login screen after Jetpack Setup succeeds:
- Fetch the email address from the Jetpack authorize URL query param is possible - and send this in the completion closure when Jetpack Setup succeeds.
- Use the authorized email address to start the WP.com login flow.
- Update the `currentSelfHostedSite` for valid Jetpack connection after the setup. This is necessary since the custom login flow doesn't have information about the site URL, so the `currentSelfHostedSite` is needed to infer the site address and navigate the user to the home screen immediately without showing the site picker.

### Testing instructions

- Prerequisite: a self-hosted with WooCommerce and without Jetpack.
- Open the app, and log out if necessary. On the prologue screen, select Enter a site address
- Enter your site address, then proceed to Sign in with site credentials.
- Enter the username and password for your WP.org account and tap continue.
- Notice that the Jetpack error screen is then displayed. Select Install Jetpack. Follow the steps to set up Jetpack. 
- After the process is done, notice that a new screen is presented asking for credentials to WP.com. Notice that the email used for the Jetpack authorization step has been prefilled in the email field.
- Enter the password for your WP.com account. After the login succeeds, notice that the app navigates to the home screen.

Please feel free to test again with Site Address + WP.com login flow. Everything should work as before.

### Screenshots

No recording here since the flow includes sensitive information.

---
- [ ] Update pod version for `WordPressAuthenticator` after https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/653 is merged and released.
